tasks/code_sync_status.rb

```diff
@@ -0,0 +1,93 @@
+#!/opt/puppetlabs/puppet/bin/ruby
+
+# Puppet Task Name: code_sync_status
+require 'net/https'
+require 'uri'
+require 'json'
+require 'puppet'
+
+# CodeSyncStatus task class
+class CodeSyncStatus
+  def initialize(params)
+    @params = params
+  end
+
+  def execute!
+    puts syncstatus.to_json
+  end
+
+  def https
+    https = Net::HTTP.new('localhost', '8140')
+    https.use_ssl = true
+    https.cert = @cert ||= OpenSSL::X509::Certificate.new(File.read(Puppet.settings[:hostcert]))
+    https.key = @key ||= OpenSSL::PKey::RSA.new(File.read(Puppet.settings[:hostprivkey]))
+    https.verify_mode = OpenSSL::SSL::VERIFY_NONE
+    https
+  end
+
+  def apistatus
+    status = https
+    # Only debug level includes code sync details
+    status_request = Net::HTTP::Get.new('/status/v1/services?level=debug')
+    JSON.parse(status.request(status_request).body)
+  end
+
+  def checkenvironmentlist(environments, requestedenvironments)
+    environmentstocheck = []
+    # If all was passed as an argument we check all visible environments
+    if requestedenvironments.any? { |s| s.casecmp('all') == 0 }
+      environmentstocheck = environments
+    # Else check each requested environment to confirm its a visible environment
+    else
+      requestedenvironments.each do |environment|
+        environments.any? { |s| s.casecmp(environment.to_s) == 0 } || raise("Environment #{environment} is not visible and will not be checked")
+        environmentstocheck << environment
+      end
+    end
+    environmentstocheck
+  end
+
+  def checkenvironmentcode(environment, servers, statuscall)
+    # Find the commit ID of the environment according to the file sync service
+    primarycommit = statuscall['file-sync-storage-service']['status']['repos']['puppet-code']['submodules'][environment.to_s]['latest_commit']['message'][32..71]
```

Comment:
Just wondering, why do we need only the last 8 digits? (`[32..71]`)

What if ... for whatever reason the commit-id changes in size?

My suggestion will be to just pick the last 8 digits (if we really want just the last 8) by using:

```
# This will work regardless of the size of the string
...['message'][-8,8]
```

Reply:
So the api returns this value (rather unhelpfully) below 

code-manager deploy signature: '93027145096d9f1e0b716b20b8129618d0a2c7e2'

Which shouldn't change but means i am taking from the 32nd to the 73rd to avoid the 74th character quote